### PR TITLE
feat(artifact): add retrieve messages and write message tasks

### PR DESCRIPTION
**Pull request by chuang8511**
Because

- we want catalog to able to have the memory system to achieve AI assistant

This commit

- implement the tasks about the messages

### CODE REVIEW

**Comment by chuang8511** [comment]
There are 2 things that we have to check and do
1. e2e test
2. test code

File: data/artifact/v0/config/tasks.json [hunk 1]
```diff
@@ -1029,36 +1096,338 @@
...
+              "update-time"
+            ],
+            "title": "Message",
+            "type": "object"
+          },
+          "instillFormat": "array:object",
+          "title": "Messages",
+          "type": "array"
+        },
+        "next-page-token": {
```
Comment (donch1989) [inline data/artifact/v0/config/tasks.json:1250]: We don't have page-token in input, there is no need to return next-page-token.

File: data/artifact/v0/config/tasks.json [hunk 2]
```diff
@@ -1029,36 +1096,338 @@
...
+          "items": {
+            "properties": {
+              "message-uid": {
+                "description": "The unique identifier of the message",
+                "instillFormat": "string",
+                "instillUIOrder": 0,
+                "title": "Message ID",
+                "type": "string"
+              },
+              "catalog-id": {
```
Comment (donch1989) [inline data/artifact/v0/config/tasks.json:1183]: I think all messages in the array should share the same `catalog-id` and `conversation-id`, also matching the user's input. There's no need to return them in the output again.

**Review by donch1989 — CHANGES_REQUESTED** [review]

**Comment by chuang8511** [comment]
Convert it to draft because we will modify it after standardised AI task.

File: data/artifact/v0/chat_operation.go [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+	artifactClient, connection := e.client, e.connection
+	defer connection.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	ctx = metadata.NewOutgoingContext(ctx, getRequestMetadata(e.SystemVariables))
+
+	_, err = artifactClient.CreateConversation(ctx, &artifactPB.CreateConversationRequest{
```
Comment (chuang8511) [inline data/artifact/v0/chat_operation.go:199]: please use ListConversations.
If there is no Conversation, call CreateConversation.

**Pull request closed without merging**